pkg/cri/server/container_stats_list_linux.go

```diff
@@ -64,22 +103,64 @@ func (c *criService) containerMetrics(
 			return nil, fmt.Errorf("failed to extract container metrics: %w", err)
 		}
 
-		cpuStats, err := c.cpuContainerStats(meta.ID, false /* isSandbox */, s, protobuf.FromTimestamp(stats.Timestamp))
+		c.setContainerStats(meta.ID, false, &cs)
```

Comment:
It's odd to store a pointer to a struct, and then continue to mutate the struct (lines 115 and 121).  It looks like this breaks the intent of the mutex in the `sandbox.Store` and `container.Store`, opening up the possibility of a data race.